library/cisco_imc_ipmi.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python
+
+from ansible.module_utils.basic import *
+
+DOCUMENTATION = '''
+---
+module: cisco_imc_ipmi
+short_description: Configures ipmi a Cisco IMC Server
+version_added: "0.9.0.0"
+description:
+    - Configures the Serial Over Lan(SOL) service on a Cisco IMC Server
+Input Params:
+    priv:
+        description: Privilege to be used
+        required: False
+        choices: ["admin", "user", "read-only"]
+        default: 'user'
+
+    key:
+        description: Hexadecimal Key to be used for authentication
+        required: False
+
+    server_id:
+        description: Server Id to be specified for C3260 platforms
+        choices: ["admin", "read-only", "user"]
+        default: "read-only"
+        required: False
+
+    state:
+        description: Used to create or delete the SOL console
+        choices: ["present", "absent"]
+        default: "present"
+        required: False
+
+requirements: ['imcsdk']
+author: "Branson Matheson (brmathes@cisco.com)"
+'''
+
+EXAMPLES = '''
+- name: enable IPMI
+  cisco_imc_ipmi:
+    priv: "admin"
+    state: "present"
+    ip: "192.168.1.1"
+    username: "admin"
+    password: "password"
+'''
+
+
+def login(module):
+    ansible = module.params
+    server = ansible.get('server')
+    if server:
+        return server
+
+    from imcsdk.imchandle import ImcHandle
+    results = {}
+    try:
+        server = ImcHandle(ip=ansible["ip"],
+                           username=ansible["username"],
+                           password=ansible["password"],
+                           port=ansible["port"],
+                           secure=ansible["secure"],
+                           proxy=ansible["proxy"])
+        server.login()
+    except Exception as e:
+        results["msg"] = str(e)
+        module.fail_json(**results)
+    return server
+
+
+def logout(module, imc_server):
+    ansible = module.params
+    server = ansible.get('server')
+    if server:
+        # we used a pre-existing handle from another task.
+        # do not logout
+        return False
+
+    if imc_server:
+        imc_server.logout()
+        return True
+    return False
+
+
+def setup_ipmi(server, module):
+    from imcsdk.apis.admin.ipmi import ipmi_disable
+    from imcsdk.apis.admin.ipmi import ipmi_enable
+    from imcsdk.apis.admin.ipmi import is_ipmi_enabled
```

Comment:
@waghswapnil  this is more of a comment on API layer provided by imcsdk - `is_ipmi_enabled` -> `ipmi_is_enabled`